framework/test/mocha/functional/http/get/transactions.js

```diff
@@ -356,6 +357,120 @@ describe('GET /api/transactions', () => {
 						});
 					});
 			});
+
+			describe('asset field', () => {
+				it('using type 0 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: TRANSACTION_TYPES.SEND }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction =>
+								expect(Object.keys(transaction.asset).length).to.be.below(2)
+							);
+						});
+				});
+
+				it('using type 1 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: TRANSACTION_TYPES.SIGNATURE }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction => {
+								expect(transaction.asset).to.have.property('signature');
+								expect(transaction.asset.signature).to.have.property(
+									'publicKey'
+								);
+								return expect(transaction.asset.signature.publicKey).to.be.a(
+									'string'
+								);
+							});
+						});
+				});
+
+				it('using type 2 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: TRANSACTION_TYPES.DELEGATE }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction => {
+								expect(transaction.asset.delegate).to.have.property(
+									'publicKey'
+								);
+								expect(transaction.asset.delegate).to.have.property('username');
+								return expect(transaction.asset.delegate).to.have.property(
+									'address'
+								);
+							});
+						});
+				});
+
+				it('using type 3 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: TRANSACTION_TYPES.VOTE }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+							// Skip Genesis vote transaction - exception as it contains 101 votes
+							const transactionsType3 = res.body.data.filter(
+								transaction =>
+									transaction.recipientId !== '16313739661670634666L'
+							);
+							transactionsType3.map(transaction => {
```

Comment:
Add an expectation like `expect(transactionsType3.length).to.greaterThan(0)` to make sure we have data to test the next line.

Reply:
@lsilvs expect(res.body.data).to.not.empty; is kind of ensuring that tho @michielmulders I personally prefer to check length as well because empty matcher can be used for strings as well.

Reply:
@yatki  But after `expect(res.body.data).to.not.empty` the `res.body.data` is being filtered meaning it can result in an empty array.